planner/core/task.go

```diff
@@ -1142,6 +1190,7 @@ func (p *PhysicalLimit) attach2Task(tasks ...task) task {
 			pushedDownLimit.SetSchema(pushedDownLimit.children[0].Schema())
 			pushedDownLimit.cost = cop.cost()
 		}
+		cop.expectCnt = p.Count
```

Comment:
Better to use `PhysicalProperty.ExpectedCnt` to set `cop.expectCnt` in `DataSource.convertToIndexScan`.